### ISSUE TITLE
deprecate load_setuptools_entrypoints

### ISSUE THREAD

**Issue by RonnyPfannschmidt** [pytest-dev/pluggy#154]
entry-point loading should be moved into a interface where one passes the entry-point system (either pkg_resources or entry_points), the plugin-manager and the entrypoint name into a helper that facilitates registration

**Comment by nicoddemus** [comment]
Good idea, this will make `pluggy` more useful in contexts outside the setuptools ecosystem.

**Comment by RonnyPfannschmidt** [comment]
@tgoodlet im not aware of any - the distribution<>entryoint mapping things is just something that is generalizable as a next step

**Comment by anxuae** [comment]
Hello,

I'm wondering if it is possible that the helper can load a plugin from a path to a python module?
This will help to load plugin form user who don't fully mastering in packaging using setuptools.

for instance a function `load_entrypoints(pm, entrypoint)`, the `entrypoint` could be a key has defined in setuptools or a path to a python module to be loaded.

The module [`imp`](https://docs.python.org/3/library/imp.html) could be used to load the module.

BR
anxuae
